go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -668,16 +677,23 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		// IP/MAC from the annotation.
 		lsp.DynamicAddresses = nil
 
-		if bnc.doesNetworkRequireIPAM() {
-			// ensure we have reserved the IPs in the annotation
-			if err = bnc.lsManager.AllocateIPs(switchNames.Original, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
-				return nil, nil, nil, false, fmt.Errorf("unable to ensure IPs allocated for already annotated pod: %s, IPs: %s, error: %v",
-					podDesc, util.JoinIPNetIPs(podIfAddrs, " "), err)
-			} else {
-				needsIP = false
+		// At interconnect the logical switch manager only knows stuff about
+		// the current switch if we ask for original switch after
+		// live migration this fails, that's why we skip it here.
+		if switchNames.Original != switchNames.Current {
```

Comment:
Also, this is a bit more nuanced than I thought.

Basically we are counting on the fact that 
- the IPs were already allocated for the original pod
- but on restart, the IPs will be allocated in syncPods/allocateIPs for this pod as well

so that we don't need to allocate them again now. I would be very explicit about this in the comment.